cpu/esp_common/syscalls.c

```diff
@@ -82,13 +82,16 @@ void IRAM_ATTR _lock_init(_lock_t *lock)
 {
     assert(lock != NULL);
 
+    uint32_t state = irq_disable();
     mutex_t* mtx = malloc(sizeof(mutex_t));
 
     if (mtx) {
         memset(mtx, 0, sizeof(mutex_t));
         *lock = (_lock_t)mtx;
     }
     /* cppcheck-suppress memleak; mtx is stored in lock */
+
+    irq_restore(state);
```

Comment:
I think it would be safe to move this up directly after the call to `malloc()`

Reply:
Hm, as long as `mtx` is not assigned `*lock`, the locking variable looks to be uninitialized. That is, another thread could try to reinitialize the same variable if a context switch occurs after `malloc` what might be the case if interrupts are pending and the interrupts are enabled. This wouldn't be a problem if all locking variables were initialized before scheduling becomes active. Unfortunately, most locking variables in the newlib are implicitly initialized when `_lock_acquire_*` is called the first time. That's why I guarded the whole function.